src/game.rs

```diff
@@ -11,37 +11,33 @@ pub struct Game {
 impl Game {
     pub fn new() -> Game {
         // initial bank cards
-        let mut dealer_cards: Vec<Card> = Vec::new();
-        dealer_cards.push(Card::random());
-        dealer_cards.push(Card::random());
+        let dealer_cards: Vec<Card> = vec![Card::random(), Card::random()];
         let dealer_score: u8 = Game::compute_score(&dealer_cards);
-        
+
         // initial player cards
-        let mut player_cards: Vec<Card> = Vec::new();
-        player_cards.push(Card::random());
-        player_cards.push(Card::random());
+        let player_cards: Vec<Card> = vec![Card::random(), Card::random()];
 
         let player_score: u8 = Game::compute_score(&player_cards);
         // return randomly initialized game state
         Game {
-            dealer_cards: dealer_cards,
-            dealer_score: dealer_score,
-            player_cards: player_cards,
-            player_score: player_score,
+            dealer_cards,
+            dealer_score,
+            player_cards,
+            player_score,
         }
     }
 
     pub fn player_draw_card(&mut self) {
         self.player_cards.push(Card::random());
         self.player_score = Game::compute_score(&self.player_cards);
     }
-    
+
     pub fn dealer_draw_card(&mut self) {
         self.dealer_cards.push(Card::random());
         self.dealer_score = Game::compute_score(&self.dealer_cards);
     }
-    
-    fn compute_score(cards: &Vec<Card>) -> u8 {
+
+    fn compute_score(cards: &[Card]) -> u8 {
```

Comment:
@billmakes do you understand this shortcut?

Reply:
Uhh sort of lol. So from my understanding, explicitly requiring a whole reference to a vector collection of type Card restricts the flexibility of your `compute_score` function. Allowing any slice containing Card elements will let your function not only just take in a whole vector collection but any slice of Card elements as well. I'd say this is probably not a big deal at all for the scope of this project but clippy will let you know how it feels either way, it couldn't hurt to keep in mind. There's also something about indirection and how the references are pointing to the memory addresses but it's a bit beyond me haha

I actually recently learned this from the following resources: 

https://rust-lang.github.io/rust-clippy/master/
https://rust-unofficial.github.io/patterns/idioms/coercion-arguments.html#description
https://www.youtube.com/watch?v=hBA00Tx_dIM